### ISSUE TITLE
Data Transforms: Masking and Distortion

### ISSUE THREAD

**Issue by fluegelk** [Helmholtz-AI-Energy/VOLTA#50]
Different masking and distortion strategies should be applied as transforms in the `transform` of the dataset
- inherit from Transforms class
- pass list of transforms to dataset class

Tasks
- [ ] Define an interface for graph transforms, take inspiration from pytorch, torchvision,...
- [ ] Random masking for training
	- fixed masking value, default 0
	- masking probability: mask each position with probability x
	- potential alternative masking strategy: mask x percent of random positions
- [ ] Distortion: distort the values at random positions
	- can reuse position selection from masking
	- instead of setting a fixed value, distort/change the value → how exactly?
- [ ] Zero-shot masking: mask entire nodes or timesteps
	- [ ] Node masking: mask the features of an entire (random) node
	- [ ] Time masking: mask one or multiple entire timesteps (node and edge features)
		- either random timestep or at the end
